sim/midas/src/main/scala/midas/models/dram/Util.scala

```diff
@@ -410,9 +410,9 @@ class FIFOAlignedAddressMatcher(val entries: Int, addrWidth: Int, size: Int, len
 
   io.hit := addrs.exists({entry =>  entry.valid && 
                                     ((entry.bits.addrmin <= inAddrMin && 
-                                      inAddrMin < entry.bits.addrmax) ||  
-                                    (inAddrMax <= entry.bits.addrmax && 
-                                      entry.bits.addrmin < inAddrMax))})
+                                      inAddrMin <= entry.bits.addrmax) ||  
```

Comment:
This is longer than it needs to be. See my previous comment or this SO post:
https://stackoverflow.com/questions/325933/determine-whether-two-date-ranges-overlap/325964#325964

Reply:
Has only two comparisons